cmake/modules/RootConfiguration.cmake

```diff
@@ -642,6 +642,10 @@ set(ROOT_BINARY_DIR ${CMAKE_BINARY_DIR}/bin)
 ")
 set(ROOT_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake/modules")
 
+# We need to overwrite ROOT_BINARY_DIR automatically generated by CMake and pointing to
+# ${CMAKE_BINARY_DIR}.
+set(ROOT_BINARY_DIR ${CMAKE_BINARY_DIR}/bin CACHE INTERNAL "")
```

Comment:
`${ROOT_BINARY_DIR}` is literally the directory where you called CMake from, it cannot be adjusted or things may start failing. What you are probably after is `root-config --bindir`, which is not the same thing.

Reply:
So then there is mistake in ROOTConfig.cmake? https://gist.github.com/oshadura/d99a6b3c8ead2cd3cccc81517e5f914d

Reply:
Just to write down what we discussed in person... Yes, we should rename `ROOT_BINARY_DIR` from `ROOTConfig.cmake`, since it has a different meaning than the above, and for roottest it needs it to find `root.exe` both as part of ROOT and as a standalone project. Ideally, though, we should update `ROOTConfig.cmake` to export all configurable directories, as now the only way to find out is using `root-config`. At the very least, we should export a variable pointing to the `root-config` shell script, but that doesn't work on Windows, so exporting things in CMake is the way to go for later.